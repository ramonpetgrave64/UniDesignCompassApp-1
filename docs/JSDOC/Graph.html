<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Graph</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Graph</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Graph<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description">Component that accepts a processId, fetches its data and displays its information</div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="Graph"><span class="type-signature"></span>new Graph<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Graph.js.html">components/Graph.js</a>, <a href="components_Graph.js.html#line15">line 15</a>
    </li></ul></dd>
    

    

    

    
</dl>




















    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="bar_chart_render"><span class="type-signature"></span>bar_chart_render<span class="type-signature"></span></h4>




<div class="description">
    Renders the ResponsiveBarChart for the loaded chart data in state
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Graph.js.html">components/Graph.js</a>, <a href="components_Graph.js.html#line201">line 201</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="deleteLogHandler"><span class="type-signature"></span>deleteLogHandler<span class="type-signature"></span></h4>




<div class="description">
    Sends a request to delete a Log, using Utils.deleteLogs()
Reloads all Logs again
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Graph.js.html">components/Graph.js</a>, <a href="components_Graph.js.html#line127">line 127</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="loading_render"><span class="type-signature"></span>loading_render<span class="type-signature"></span></h4>




<div class="description">
    Displays a spinner in the middle of the screen, while the loading key in state is true
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Graph.js.html">components/Graph.js</a>, <a href="components_Graph.js.html#line209">line 209</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="process_logs_render"><span class="type-signature"></span>process_logs_render<span class="type-signature"></span></h4>




<div class="description">
    Renders a LogCard for each of the Logs loaded into state
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Graph.js.html">components/Graph.js</a>, <a href="components_Graph.js.html#line160">line 160</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="updateLogHandler"><span class="type-signature"></span>updateLogHandler<span class="type-signature"></span></h4>




<div class="description">
    Sends a request to update a Log, using Utils.updateLogs()
Reloads all Logs again.
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Graph.js.html">components/Graph.js</a>, <a href="components_Graph.js.html#line145">line 145</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="load_chart_data"><span class="type-signature"></span>load_chart_data<span class="signature">(process_id)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Loads the Phase duration data for the BarChart and saves to state
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>process_id</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The id of the Process</td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Graph.js.html">components/Graph.js</a>, <a href="components_Graph.js.html#line65">line 65</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id="load_log_data"><span class="type-signature"></span>load_log_data<span class="signature">(process_id)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Loads the Phase Logs data and saves to state
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>process_id</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The id of the Process</td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Graph.js.html">components/Graph.js</a>, <a href="components_Graph.js.html#line92">line 92</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id="load_process_data"><span class="type-signature"></span>load_process_data<span class="signature">(process_id)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Loads the data related to the BarChart and data related to the Logs in a Process
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>process_id</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The id of the Process</td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Graph.js.html">components/Graph.js</a>, <a href="components_Graph.js.html#line28">line 28</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id="msToHours"><span class="type-signature"></span>msToHours<span class="signature">(t)</span><span class="type-signature"> &rarr; {number}</span></h4>
    

    



<div class="description">
    Converts a number in milliseconds to hours
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>t</code></td>
            

            <td class="type">
            
                
<span class="param-type">number</span>


            
            </td>

            

            

            <td class="description last">The number of milliseconds</td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Graph.js.html">components/Graph.js</a>, <a href="components_Graph.js.html#line53">line 53</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    The milliseconds converted to hours
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">number</span>


    </dd>
</dl>

    





        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="ChooseStructure.html">ChooseStructure</a></li><li><a href="Compass.html">Compass</a></li><li><a href="CreatePage.html">CreatePage</a></li><li><a href="Dashboard.html">Dashboard</a></li><li><a href="DescribePhase.html">DescribePhase</a></li><li><a href="Graph.html">Graph</a></li><li><a href="LogCard.html">LogCard</a></li><li><a href="Login.html">Login</a></li><li><a href="NewPassword.html">NewPassword</a></li><li><a href="Phase.html">Phase</a></li><li><a href="PhaseNumber.html">PhaseNumber</a></li><li><a href="Process.html">Process</a></li><li><a href="SignUp.html">SignUp</a></li><li><a href="SubmitCompass.html">SubmitCompass</a></li><li><a href="Verification.html">Verification</a></li></ul><h3>Global</h3><ul><li><a href="global.html#LogHandler">LogHandler</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.2</a> on Thu May 23 2019 18:37:53 GMT-0400 (Eastern Daylight Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>