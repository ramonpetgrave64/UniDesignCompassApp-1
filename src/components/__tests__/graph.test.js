import React from "react";
import { create } from "react-test-renderer";
import Graph from "../Graph";
import Adapter from 'enzyme-adapter-react-16';
import {configure, mount, shallow} from 'enzyme';

configure({adapter:new Adapter()});
let GraphWrapper=shallow(<Graph/>)

describe("Graph", () => {
  it("it matches the snapshot", () => {
    const component = create(
    <Graph 
        processId={'ec468ab1-9ed8-41fe-a88a-3c85cdbc7fc6'}
    />);
    expect(component.toJSON()).toMatchSnapshot();
  });


it('Constructor called properly.', () => {
  let spy=jest.spyOn(GraphWrapper.instance(),"constructor")
  expect(spy).toHaveBeenCalledTimes(0);
  });


it('componentDidMount ', () => {
  let spy=jest.spyOn(GraphWrapper.instance(),"componentDidMount")
  expect(spy).toHaveBeenCalledTimes(0);
  });

it('componentDidMount test.', () => {
  let spy=jest.spyOn(GraphWrapper.instance(),"componentDidMount")
  expect(spy).toHaveBeenCalledTimes(0);
  GraphWrapper.instance().componentDidMount()
  expect(spy).toHaveBeenCalledTimes(1);
  });

it('componentDidUpdate ',()=>{
let spy=jest.spyOn(GraphWrapper.instance(),"componentDidUpdate")
expect(spy).toHaveBeenCalledTimes(0);
});  

it('componentDidUpdate test.', () => {
  let spy=jest.spyOn(GraphWrapper.instance(),"componentDidUpdate")
  expect(spy).toHaveBeenCalledTimes(0);
  let prevProps
  GraphWrapper.instance().componentDidUpdate(prevProps)
  expect(spy).toHaveBeenCalledTimes(1);
  });

  it('msToHours ', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"msToHours")
    expect(spy).toHaveBeenCalledTimes(0);
    });
  
  it('msToHours test', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"msToHours")
    expect(spy).toHaveBeenCalledTimes(0);
    spy(10000)
    expect(spy).toHaveBeenCalledTimes(1);
    expect(Math.floor(spy(1000))).toEqual(0);
    });
    
  it('load_chart_data ', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"load_chart_data")
    expect(spy).toHaveBeenCalledTimes(0);
    });  
  
  it('load_chart_data test', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"load_chart_data")
    expect(spy).toHaveBeenCalledTimes(0);
    GraphWrapper.instance().load_chart_data()
    expect(spy).toHaveBeenCalledTimes(1);
    });
  
  it('load_log_data ', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"load_log_data")
    expect(spy).toHaveBeenCalledTimes(0);
    });  
  
  it('load_log_data test', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"load_log_data")
    expect(spy).toHaveBeenCalledTimes(0);
    GraphWrapper.instance().load_log_data()
    expect(spy).toHaveBeenCalledTimes(1);
    });

  it('deleteLogHandler ', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"deleteLogHandler")
    expect(spy).toHaveBeenCalledTimes(0);
    }); 

  it('deleteLogHandler test', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"deleteLogHandler")
    expect(spy).toHaveBeenCalledTimes(0);
    GraphWrapper.instance().deleteLogHandler()
    expect(spy).toHaveBeenCalledTimes(1);
    });

  it('updateLogHandler ', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"updateLogHandler")
    expect(spy).toHaveBeenCalledTimes(0);
    });

  it('updateLogHandler test', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"updateLogHandler")
    expect(spy).toHaveBeenCalledTimes(0);
    GraphWrapper.instance().updateLogHandler()
    expect(spy).toHaveBeenCalledTimes(1);
    });

  it('process_logs_render ', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"process_logs_render")
    expect(spy).toHaveBeenCalledTimes(0);
    });
    

  it('process_logs_render test', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"process_logs_render")
    expect(spy).toHaveBeenCalledTimes(0);
    GraphWrapper.instance().process_logs_render()
    expect(spy).toHaveBeenCalledTimes(1);
    });

  it('bar_chart_render ', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"bar_chart_render")
    expect(spy).toHaveBeenCalledTimes(0);
  });
      
  it('bar_chart_render test', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"bar_chart_render")
    expect(spy).toHaveBeenCalledTimes(0);
    GraphWrapper.instance().bar_chart_render()
    expect(spy).toHaveBeenCalledTimes(1);
    });
  
  it('loading_render ', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"loading_render")
    expect(spy).toHaveBeenCalledTimes(0);
  });
      
  it('loading_render test', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"loading_render")
    expect(spy).toHaveBeenCalledTimes(0);
    GraphWrapper.instance().loading_render()
    expect(spy).toHaveBeenCalledTimes(1);
    });

  it('render ', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"render")
    expect(spy).toHaveBeenCalledTimes(0);
  });
      
  it('render test', () => {
    let spy=jest.spyOn(GraphWrapper.instance(),"render")
    expect(spy).toHaveBeenCalledTimes(0);
    GraphWrapper.instance().render()
    expect(spy).toHaveBeenCalledTimes(1);
    });
  

});
