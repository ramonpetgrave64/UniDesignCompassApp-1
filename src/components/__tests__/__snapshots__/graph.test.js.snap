// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graph it matches the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="spinner-grow text-info"
    style={
      Object {
        "height": "5em",
        "left": "50%",
        "marginLeft": "-2.5em",
        "marginTop": "-2.5em",
        "position": "fixed",
        "top": "50%",
        "width": "5em",
        "zIndex": "1020",
      }
    }
  />
</div>
`;

exports[`it matches the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="spinner-grow text-info"
    style={
      Object {
        "height": "5em",
        "left": "50%",
        "marginLeft": "-2.5em",
        "marginTop": "-2.5em",
        "position": "fixed",
        "top": "50%",
        "width": "5em",
        "zIndex": "1020",
      }
    }
  />
</div>
`;
